src/ai/backend/web/proxy.py

```diff
@@ -204,26 +200,9 @@ async def web_handler(request: web.Request, *, is_anonymous=False) -> web.Stream
                 api_rqst.headers["Content-Length"] = request.headers["Content-Length"]
             if "Content-Length" in request.headers and secure_context:
                 api_rqst.headers["Content-Length"] = str(decrypted_payload_length)
-            for hdr in HTTP_HEADERS_TO_FORWARD:
+            for hdr in {*HTTP_HEADERS_TO_FORWARD, *extra_forwarding_headers}:
                 if request.headers.get(hdr) is not None:
                     api_rqst.headers[hdr] = request.headers[hdr]
```

Comment:
The header seems to allow being overridden, which poses a potential security risk.

Reply:
Resolves by 143cc14d17fff7c537cb75ecdfd1ae5dac2e5d31.